docs/Inside_Arbitrum_Nitro/Inside_Arbitrum_Nitro.md

```diff
@@ -0,0 +1,527 @@
+
+Being able to set the speed limit accurately depends on being able to estimate the time required to validate an RBlock, with some accuracy. Any uncertainty in estimating validation time will force us to set the speed limit lower, to be safe. And we do not want to set the speed limit lower, so we try to enable accurate estimation.
+
+### Fees
+
+User transactions pay fees, to cover the cost of operating the chain. These fees are assessed and collected by ArbOS at L2. They are denominated in ETH. 
+
+Fees are charged for two resources that a transaction can use:
+
+* *L2 gas*: an Ethereum-equivalent amount of gas, as required to execute the transaction on the Nitro chain,
+
+- _L1 calldata_: a fee per unit of L1 calldata attributable to the transaction, which is charged only if the transaction came in via the sequencer, and is paid to the sequencer to cover its costs,
+
+#### L2 gas fees
+
+L2 gas fees work very similarly to gas on Ethereum. A transaction uses some amount of gas, and this is multiplied by the current basefee to get the L2 gas fee charged to the transaction. 
+
+The L2 basefee is set by a version of the "exponential mechanism" which has been widely discussed in the Ethereum community, and which has been shown equivalent to Ethereum's EIP-1559 gas pricing mechanism. 
+
+The algorithm compares gas usage against a parameter called the "speed limit" which is the target amount of gas per second that the chain can handle sustainably over time. (Currently the speed limit is 7 million gas per second.) The algorithm tracks a gas backlog. Whenever a transaction consumes gas, that gas is added to the backlog. Whenever the clock ticks one second, the speed limit is subtracted from the backlog; but the backlog can never go below zero. 
+
+Intuitively, if the backlog grows, the algorithm should increase the gas price, to slow gas usage, because usage is above the sustainable level. If the backlog shrinks, the price should decrease again because usage has been below the below the sustainable limit so more gas usage can be welcomed. 
+
+To make this more precise, the basefee is an exponential function of the backlog, *F = exp(-a(B-b))*, where a and b are suitably chosen constants: *a* controls how rapidly the price escalates with backlog, and *b* allows a small backlog before the basefee escalation begins.
+
+#### L1 calldata fees
+
+L1 calldata fees exist because the sequencer, or the batch poster which posts the sequencer's transaction batches on Ethereum,  incurs costs in L1 gas to post transactions on Ethereum as calldata.  Funds collected in L1 calldata fees are credited to the batch poster to cover its costs. 
```

Comment:
```suggestion
L1 calldata fees exist because the sequencer, or the batch poster which posts the sequencer's transaction batches on Ethereum, incurs costs in L1 gas to post transactions on Ethereum as calldata. Funds collected in L1 calldata fees are credited to the batch poster to cover its costs. 
```